AEPMessaging/Tests/UnitTests/Messaging+PublicApiTest.swift

```diff
@@ -119,4 +119,107 @@ class MessagingPublicApiTest: XCTestCase {
+        let trigger = UNCalendarNotificationTrigger(dateMatching: dateInfo, repeats: false)
+        let notificationContent = UNMutableNotificationContent()
+        
+        let request = UNNotificationRequest(identifier: mockIdentifier, content: notificationContent, trigger: trigger)
+        guard let response = UNNotificationResponse(coder: MockNotificationResponseCoder(with: request)) else {
+            XCTFail()
+            return
+        }
+        
+        Messaging.handleNotificationResponse(response, applicationOpened: true, customActionId: mockCustomActionId)
+        wait(for: [expectation], timeout: 1)
+    }
+    
+    func testHandleNotificationResponseEmptyMessageId() {
+        let expectation = XCTestExpectation(description: "Messaging request event")
+        let mockCustomActionId = "mockCustomActionId"
+        let mockIdentifier = ""
+        expectation.assertForOverFulfill = true
+        expectation.isInverted = true
+        
+        EventHub.shared.getExtensionContainer(MockExtension.self)?.eventListeners.clear()        
+        EventHub.shared.getExtensionContainer(MockExtension.self)?.registerListener(type: MessagingConstants.Event.EventType.messaging, source: EventSource.requestContent) { event in
+            XCTFail()
+            expectation.fulfill()
+        }
+        
+        let dateInfo = Calendar.current.dateComponents([.year, .month, .day, .hour, .minute, .second], from: Date())
+        let trigger = UNCalendarNotificationTrigger(dateMatching: dateInfo, repeats: false)
+        let notificationContent = UNMutableNotificationContent()
+        
+        let request = UNNotificationRequest(identifier: mockIdentifier, content: notificationContent, trigger: trigger)
+        guard let response = UNNotificationResponse(coder: MockNotificationResponseCoder(with: request)) else {
+            XCTFail()
+            return
+        }
+        
+        Messaging.handleNotificationResponse(response, applicationOpened: true, customActionId: mockCustomActionId)
+        wait(for: [expectation], timeout: 1)
+    }
+    
+    func testRefreshInAppMessages() throws {
+        // setup
+        let expectation = XCTestExpectation(description: "Refresh In app messages event")
+        expectation.assertForOverFulfill = true
+        
+        EventHub.shared.getExtensionContainer(MockExtension.self)?.eventListeners.clear()
+        EventHub.shared.getExtensionContainer(MockExtension.self)?.registerListener(type: MessagingConstants.Event.EventType.messaging, source: EventSource.requestContent) { event in
+            XCTAssertEqual(MessagingConstants.Event.Name.REFRESH_MESSAGES, event.name)
+            XCTAssertEqual(MessagingConstants.Event.EventType.messaging, event.type)
+            XCTAssertEqual(EventSource.requestContent, event.source)
+            
+            guard let eventData = event.data else {
+                XCTFail()
+                expectation.fulfill()
```

Comment:
can we remove this expectation.fulfill() ?

Reply:
it's a short-circuit on the expectation so we don't `wait` the full duration